c++/src/kj/memory.h

```diff
@@ -87,39 +87,26 @@ using RefOrVoid = typename RefOrVoid_<T>::Type;
 //
 // This is a hack needed to avoid defining Own<void> as a totally separate class.
 
-template <typename T, bool isPolymorphic = _kj_internal_isPolymorphic((T*)nullptr)>
-struct CastToVoid_;
-
-template <typename T>
-struct CastToVoid_<T, false> {
-  static void* apply(T* ptr) {
-    return static_cast<void*>(ptr);
-  }
-  static const void* applyConst(T* ptr) {
-    const T* cptr = ptr;
-    return static_cast<const void*>(cptr);
-  }
-};
-
 template <typename T>
-struct CastToVoid_<T, true> {
-  static void* apply(T* ptr) {
+void* castToVoid(T* ptr) {
+  if constexpr (_kj_internal_isPolymorphic((T*)nullptr)) {
     return dynamic_cast<void*>(ptr);
   }
-  static const void* applyConst(T* ptr) {
-    const T* cptr = ptr;
-    return dynamic_cast<const void*>(cptr);
+  else {
+    return static_cast<void*>(ptr);
   }
-};
-
-template <typename T>
-void* castToVoid(T* ptr) {
-  return CastToVoid_<T>::apply(ptr);
 }
 
 template <typename T>
 const void* castToConstVoid(T* ptr) {
```

Comment:
This feels like a "free lunch" simplification, hope I'm not missing anything here.

Reply:
Nope, this was just written before `if constexpr` existed and never updated.